hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -244,6 +263,18 @@ private void refreshTimeline() throws IOException {
         this.schemaProvider = srcRecordsWithCkpt.getKey();
```

Comment:
I really dislike all the nested ifs-and null checks. :( there ought to be a better way of structuring this code overall. Side rant

Reply:
Agree. the whole schema provider and write client handling needs to be refactored to be more readable.